ledger/caching/src/test/suite/scala/com/daml/caching/MappedCacheSpec.scala

```diff
@@ -0,0 +1,71 @@
+// Copyright (c) 2020 Digital Asset (Switzerland) GmbH and/or its affiliates. All rights reserved.
+// SPDX-License-Identifier: Apache-2.0
+
+package com.daml.caching
+
+import java.util.concurrent.{ConcurrentHashMap, ConcurrentMap}
+
+import org.scalatest.matchers.should.Matchers
+import org.scalatest.wordspec.AnyWordSpec
+
+final class MappedCacheSpec
+    extends AnyWordSpec
+    with Matchers
+    with CacheBehaviorSpecBase
+    with CacheCachingSpecBase {
+  override protected def name: String = "mapped cache"
+
+  override protected def newCache(): ConcurrentCache[Integer, String] =
+    new MapBackedCacheForTesting(new ConcurrentHashMap)
+
+  name should {
+    "transform the values into and out of the cache" in {
+      val store: ConcurrentMap[Int, String] = new ConcurrentHashMap
+      val cache: ConcurrentCache[Int, String] = new MapBackedCacheForTesting(store)
+      val mappedCache = cache.mapValues[String](
+        from = value => value.substring(6),
+        to = value => Some("value " + value),
+      )
+
+      mappedCache.put(7, "seven")
+      Option(store.get(7)) should be(Some("value seven"))
+      mappedCache.getIfPresent(7) should be(Some("seven"))
+    }
+
+    "allow the mapping to change type" in {
+      val store: ConcurrentMap[Int, String] = new ConcurrentHashMap
+      val cache: ConcurrentCache[Int, String] = new MapBackedCacheForTesting(store)
+      val mappedCache = cache.mapValues[Int](
```

Comment:
Please use a different `NewValue` type here (e.g., `Double` instead of `Int` as the latter is the key type of the parent cache as well).